rust/src/lib.rs

```diff
@@ -6,7 +6,13 @@ use extism_manifest::Manifest;
 mod bindings;
 
 #[repr(transparent)]
-pub struct Plugin(isize);
+pub struct Plugin(pub isize);
+
+impl Plugin {
+    pub fn as_isize(self) -> isize {
+        self.0
+    }
+}
```

Comment:
It's been a long time since I've written rust so I ran into a few obstacles. I added this so I could go from the "newtype" `Plugin` back to `isize`. See the `plugin_new_with_manifest` implementation where I cast back to isize which is automatically mapped to erlang space. I believe there is a way to implement a rustler specific trait for this type but i couldn't do it when the type is defined outside of the crate. Maybe you have a simpler solution.

Reply:
that works! I think it would be better to use the `From` trait here and implement it for both `Plugin` and `isize` bi-directionally. I have a commit to your branch incoming.